File: redsheet-portal/src/app/database-models/avoid-gesture.ts
import { required, maxLength, range } from '@rx/annotations';

export class AvoidGesture {
    constructor(avoidGesture?: AvoidGesture )  {
        let properties = [ "avoidGestureId", "avoidGestureValue", "cultureCountryId",];
        for (let property of properties)
            if (avoidGesture && avoidGesture[property])
                this[property] = avoidGesture[property];
    }
 
	avoidGestureId : number =   0 ;
 
    @required()
    @maxLength(500)
	avoidGestureValue : string =   undefined;
 
    @range(1,2147483647)
	cultureCountryId : number =   undefined;
}

File: RedSheet.DbEntities/Models/AvoidGesture.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("AvoidGestures", Schema = "dbo")]
	[Serializable]
	public partial class AvoidGesture  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int AvoidGestureId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string AvoidGestureValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int CultureCountryId { get; set; } 

	public AvoidGesture()
	{
    }

    }
}

